### DISCUSSION TITLE
Cluster performance for autoscaling

### DISCUSSION THREAD

**Discussion by abhiranjeet** [kedacore/keda#1270] [General] [answered]
Which one of these scenarios gives better performance:

1. A worker node with **20 pods with 16 GB of memory**

1. A worker node with **40 pods with 8 GB of memory**

Thanks in advance !!

**Comment by zroubalik** [comment, marked as answer]
It depends on the workloads and how do you use them. There isn't one universal answer.

**Reply by abhiranjeet** [reply]
What if they all are identical in terms of workloads ?

**Reply by tomkerkhove** [reply]
I think what @zroubalik is trying to say is that this is not something we can give a universal answer to as this highly depends on your scenario.

For example, more pods on one node is good from a cost perspective, but might be more dangerous from an availability perspective. Based on your scenario you'd choose one over the other.

That said, if those workloads are the same I think it's very unlikely that a node with 16 GB can host 20 pods while a node with 8 GB can host 40. That means the first scenario is heavily overprovisioned for the workload.

**Reply by abhiranjeet** [reply]
Ok. I understood your point. Thank You very much !
